### ISSUE TITLE
Run Syncthing integration test for Android

### ISSUE THREAD

**Issue by Zillode** [syncthing/syncthing-android#385]
```
23:25 @calmh> cd ..../syncthing/tests ; go test -v -c -tags integration
```

**Comment by lkwg82** [comment]
This might be an enhancement but very early after prototyping. Dont you ever to integration testing, even not manually? Or do you need a way to automate it?

**Comment by Zillode** [comment]
We would like to automatically bump the Android client on every Syncthing release. However, we need at least a minor test for this to see if the app still runs:)

**Comment by lkwg82** [comment]
My understanding is you need to build the app, deploy it on sth like Android phone and do some simple happy path testing. Right?

**Comment by Zillode** [comment]
Sure, you can run it on an emulator as well. We have an android jenkins build server...

**Comment by lkwg82** [comment]
What is your current testing drive? What do you like to have and have not yet (which kind of tests)?

**Comment by Zillode** [comment]
1. We need to be able to run Syncthing's integration tests on an Android device, e.g. `go test -v -c -tags integration`. But since we do not have Go natively on Android, we need a compiled version of those tests.
2. We need a small test to launch the app and see if everything works (i.e. folders and stats were fetched from the Syncthing daemon).

**Comment by Nutomic** [comment]
For the second one, check out the existing unit tests (in the `src/androidTest` folder). It shouldn't be so hard to write additional tests for SyncthingService and RestApi.

**Comment by lkwg82** [comment]
Just got the emulator with android 6 running. Started tests and noticed some tests in `SyncthingServiceTest` are broken.
![bildschirmfoto_2016-04-11_23-52-37](https://cloud.githubusercontent.com/assets/1891171/14443590/807a1158-0040-11e6-9c93-99688f4b2433.png)

Is this known or is this my poor setup?

**Comment by Nutomic** [comment]
Yeah I'm getting those too, and haven't had time to look into them. Something has probably changed in Syncthing or Android (or in syncthing-android) that broke the tests.

**Comment by lkwg82** [comment]
As I'm a really novice to Android testing plz fix that before I can continue. Just for as a guideline to every contributor.

**Comment by Nutomic** [comment]
You can just ignore those tests. But I'll try to get them fixed when I have time.

**Comment by lkwg82** [comment]
hm ... do you know "the broken window" principle? I'm not sure if I want to continue. I assume somebody invested time to create an important meaningful test. 
What about removing it? You say this test is not necessary.

**Comment by Nutomic** [comment]
Yeah I will look into it. But you're right, I should have done that sooner. It would definitely be good to have more people on the project who care about tests :)

**Comment by Catfriend1** [comment]
It would be very cool if someone experienced at writing tests on android could come up with a PR for basic functions like completing the welcome slides and configuring one example folder with test data in it. Maybe this would help Audrius daring to merge more enhancements without questioning that much if it breaks something.
